Kiosk/Auction Listings/ListingsViewController.swift

```diff
@@ -371,86 +204,21 @@ private extension ListingsViewController {
         
         return layout
     }
-    
-    // MARK: Instance methods
-    
-    func saleArtworkAtIndexPath(indexPath: NSIndexPath) -> SaleArtwork {
-        return sortedSaleArtworks[indexPath.item];
-    }
-    
-}
-
-// MARK: - Sorting Functions
-
-func leastBidsSort(lhs: SaleArtwork, _ rhs: SaleArtwork) -> Bool {
-    return (lhs.bidCount ?? 0) < (rhs.bidCount ?? 0)
-}
-
-func mostBidsSort(lhs: SaleArtwork, _ rhs: SaleArtwork) -> Bool {
-    return !leastBidsSort(lhs, rhs)
-}
-
-func lowestCurrentBidSort(lhs: SaleArtwork, _ rhs: SaleArtwork) -> Bool {
-    return (lhs.highestBidCents ?? 0) < (rhs.highestBidCents ?? 0)
-}
-
-func highestCurrentBidSort(lhs: SaleArtwork, _ rhs: SaleArtwork) -> Bool {
-    return !lowestCurrentBidSort(lhs, rhs)
 }
 
-func alphabeticalSort(lhs: SaleArtwork, _ rhs: SaleArtwork) -> Bool {
-    return lhs.artwork.sortableArtistID().caseInsensitiveCompare(rhs.artwork.sortableArtistID()) == .OrderedAscending
-}
-
-func sortById(lhs: SaleArtwork, _ rhs: SaleArtwork) -> Bool {
-    return lhs.id.caseInsensitiveCompare(rhs.id) == .OrderedAscending
-}
+// MARK: Collection view setup
 
-// MARK: - Switch Values
+extension UICollectionView {
 
-enum SwitchValues: Int {
-    case Grid = 0
-    case LeastBids
-    case MostBids
-    case HighestCurrentBid
-    case LowestCurrentBid
-    case Alphabetical
-    
-    var name: String {
-        switch self {
-        case .Grid:
-            return "Grid"
-        case .LeastBids:
-            return "Least Bids"
-        case .MostBids:
-            return "Most Bids"
-        case .HighestCurrentBid:
-            return "Highest Bid"
-        case .LowestCurrentBid:
-            return "Lowest Bid"
-        case .Alphabetical:
-            return "A–Z"
-        }
-    }
-    
-    func sortSaleArtworks(saleArtworks: [SaleArtwork]) -> [SaleArtwork] {
-        switch self {
-        case Grid:
-            return saleArtworks
-        case LeastBids:
-            return saleArtworks.sort(leastBidsSort)
-        case MostBids:
-            return saleArtworks.sort(mostBidsSort)
-        case HighestCurrentBid:
-            return saleArtworks.sort(highestCurrentBidSort)
-        case LowestCurrentBid:
-            return saleArtworks.sort(lowestCurrentBidSort)
-        case Alphabetical:
-            return saleArtworks.sort(alphabeticalSort)
-        }
-    }
-    
-    static func allSwitchValues() -> [SwitchValues] {
-        return [Grid, LeastBids, MostBids, HighestCurrentBid, LowestCurrentBid, Alphabetical]
+    class func listingsCollectionViewWithDelegateDatasource<T where T: UICollectionViewDelegate, T: UICollectionViewDataSource>(delegateDatasource: T) -> UICollectionView {
```

Comment:
![](http://media4.giphy.com/media/zH2UJxZ0OPDxu/giphy.gif)

Alright I'll do that. Thanks!